### PR TITLE
Adds an implementation of JWT signing to ServiceAccountCredentials

### PR DESCRIPTION
**Pull request by tbetbetbe**

### CODE REVIEW

**Comment by tbetbetbe** [comment]
@anthmgoogle @vsubramani  PTAL

File: lib/googleauth/service_account.rb [hunk 1]
```diff
@@ -43,6 +44,8 @@ module Auth
     #
     # cf [Application Default Credentials](http://goo.gl/mkAHpZ)
     class ServiceAccountCredentials < Signet::OAuth2::Client
```
Comment (anthmgoogle) [inline lib/googleauth/service_account.rb]: Concerns where raised about having a single type magically switch between the two very different authentication modes, particularly as the IDToken is not strictly OAuth2. While the issue is not closed a safe bet would be to make a separate class for IDToken credentials. For Java I'm planning to call this ServiceAccountIdTokenCredentials.

**Comment by anthmgoogle** [comment]
Implementation looks good. Suggest doing this as a separate class.

**Comment by tbetbetbe** [comment]
@anthmgoogle PTAL.
- it has behaviour split into it's own class, named in the convention that has been discussed
- N.B it is merged with  #17, which should be submitted before this

**Comment by tbetbetbe** [comment]
This change now also contains the promotion to from ServiceAccountCredentials to ServiceAccountJwtHeaderCredentials when the scope is set to nil

**Pull request merged**